src/cozmo/action.py

```diff
@@ -289,20 +302,35 @@ def _get_next_action_id(self):
             self.__class__._next_action_id += 1
         return next_action_id
 
-    def _send_single_action(self, action, position=0, num_retries=0):
+    @property
+    def _has_in_progress_actions(self):
+        return len(self._in_progress) > 0
+
+    def _send_single_action(self, action, in_parallel=False, num_retries=0):
         action_id = self._get_next_action_id()
         action.robot = self.robot
+        action._action_id = action_id
 
-        if len(self._in_progress) > 0:
+        if self._has_in_progress_actions and not in_parallel:
+            # Note - it doesn't matter if previous action was started as parallel,
+            # starting any subsequent action with in_parallel==False will cancel
+            # any previous actions, so we throw an exception here and require that
+            # the client explictely cancel or wait on earlier actions
```

Comment:
typo in the comment, but i appreciate it being there; preemptively answered a question ;-)